rs/ic_os/config/src/main.rs

```diff
@@ -0,0 +1,136 @@
+
+        #[arg(long, default_value_t = config::DEFAULT_SETUPOS_SSH_AUTHORIZED_KEYS_PATH.to_string(), value_name = "ssh_authorized_keys")]
+        ssh_authorized_keys_path: String,
+
+        #[arg(long, default_value_t = config::DEFAULT_SETUPOS_NODE_OPERATOR_PRIVATE_KEY_PATH.to_string(), value_name = "node_operator_private_key.pem")]
+        node_operator_private_key_path: String,
+    },
+    GenerateHostosConfig {
+        #[arg(long, default_value_t = config::DEFAULT_SETUPOS_CONFIG_OBJECT_PATH.to_string(), value_name = "config.json")]
+        setupos_config_json_path: String,
+    },
+}
+
+#[derive(Parser)]
+#[command()]
+struct ConfigArgs {
+    #[command(subcommand)]
+    command: Option<Commands>,
+}
+
+pub fn main() -> Result<()> {
+    let opts = ConfigArgs::parse();
+
+    match opts.command {
+        Some(Commands::CreateSetuposConfig {
+            config_ini_path,
+            deployment_json_path,
+            nns_public_key_path,
+            ssh_authorized_keys_path,
+            node_operator_private_key_path,
+        }) => {
+            let config_ini_path = Path::new(&config_ini_path);
+            let deployment_json_path = Path::new(&deployment_json_path);
+            // help: nns_public_key_path is copied to GuestOS (and it's contents are not copied into ic.json)
+            let nns_public_key_path = Path::new(&nns_public_key_path);
+            let ssh_authorized_keys_path = Path::new(&ssh_authorized_keys_path);
+            let node_operator_private_key_path = Path::new(&node_operator_private_key_path);
+
+            let ssh_authorized_keys_path = Some(ssh_authorized_keys_path.to_path_buf());
+
+            let node_operator_private_key_path = if node_operator_private_key_path.exists() {
+                Some(node_operator_private_key_path.to_path_buf())
+            } else {
+                None
+            };
+
+            // get config.ini variables
+            let (network_settings, verbose) = get_config_ini_settings(config_ini_path)?;
+
+            // get deployment.json variables
+            let (vm_memory, vm_cpu, nns_urls, hostname, elasticsearch_hosts) =
+                get_deployment_settings(deployment_json_path);
+
+            let icos_settings = ICOSSettings {
+                nns_public_key_path: nns_public_key_path.to_path_buf(),
+                nns_urls,
+                elasticsearch_hosts,
+                elasticsearch_tags: None,
+                hostname,
+                node_operator_private_key_path,
+                ssh_authorized_keys_path,
+            };
+
+            let hostos_settings = HostOSSettings {
+                vm_memory,
+                vm_cpu,
+                verbose,
+            };
+
+            let guestos_settings = GuestOSSettings::default();
+
+            let setupos_config = SetupOSConfig {
+                network_settings,
+                icos_settings,
+                hostos_settings,
+                guestos_settings,
```

Comment:
Do the guestos settings come directly from the setupos?

Reply:
On mainnet, yet, all guestos settings come directly from the setupOS. 

In testing, we build the guestos config directly
[rs/tests/driver/src/driver/bootstrap.rs](https://github.com/dfinity/ic/blob/e647d84f466364b5c2494c50d3a6438c0d5f452d/rs/tests/driver/src/driver/bootstrap.rs#L415) (farm)
[testnet/tools/build-guestos-configs.sh](https://github.com/dfinity/ic/blob/e647d84f466364b5c2494c50d3a6438c0d5f452d/testnet/tools/build-guestos-configs.sh#L319) (static testnets)
[rs/ic_os/launch-single-vm/src/main.rs](https://github.com/dfinity/ic/blob/e647d84f466364b5c2494c50d3a6438c0d5f452d/rs/ic_os/launch-single-vm/src/main.rs#L207) (Node team tool)

And then there's the GuestOS settings used for recovery. I'm actually not entirely sure how those are set. I'm going to open up a separate thread regarding this.